### ISSUE TITLE
v1 : MsDeploy under load can lead to host startup w/o functions

### ISSUE THREAD

**Issue by mathewc** [Azure/azure-functions-host#6248]
See ICM incident 192172973. Log analysis shows that a user initiated MsDeploy operation occurred, causing the normal host restarts as the host reacted to file change notifications. As the file system changed, the host recycled multiple times. Usually things stabilize and the final host instance restarts with a view of the newly deployed files. However, in this case, it looks like a race condition lead to the host running with no functions (there was supposed to be 1).

We need to review the file change / restart logic.

**Comment by yojagad** [comment]
@mathewc Is this related to #5151 by any chance?

**Comment by mathewc** [comment]
I don't think so - in this case the app was restarting due to file change notifications, and there were multiple host restarts. However, you could run your file system error query for this app/timerange to see if you see any of the signatures we saw before for that issue.
